### PR TITLE
Add service provider picking

### PR DESCRIPTION
**Pull request by adekbadek**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

#274 introduced handliing Constant Contact as an ESP, this PR adds UI to choose/change an ESP.

### How to test the changes in this Pull Request:

1. If you already had Mailchimp provider set up, all should work as before. If starting afresh, first set up Mailchimp on `master` branch and verify that the Mailchimp connection works as before.
2. Remove Mailchimp credentials in plugin settings (or start on a fresh site)
3. Observe the Settings view has a ESP provider selector and three credentials inputs – one for Mailchimp and two for Constant Contact
4. Create a new newsletter – observe a settings modal
5. Fill in the settings, setting up Mailchimp as the provider
6. Observe Mailchimp connection working
7. Switch the service provider to Constant Contact
7. Observe Constant Contact connection working (campaign created on newsletter creation)
7. Start from scratch (remove all options' values of start on a fresh site) on this branch, set up Constant Contact
7. Observe Constant Contact connection working
7. Switch the service provider to Mailchimp
6. Observe Mailchimp connection working

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

**Comment by adekbadek** [comment]
@jeffersonrabb – fixed in 0197decdf29d7ede1bc7123b75d760b274dd29b1

File: includes/class-newspack-newsletters.php [hunk 1]
```diff
@@ -425,37 +432,50 @@ function ( $post ) {
...
+		$credentials      = $request['credentials'];
+		$mjml_api_key     = $request['mjml_api_key'];
+		$mjml_api_secret  = $request['mjml_api_secret'];
+		$wp_error         = new WP_Error();
+
+		// Service Provider slug.
+		if ( empty( $service_provider ) ) {
+			$wp_error->add(
+				'newspack_newsletters_no_service_provider',
+				__( 'Please set at service provider.', 'newspack-newsletters' )
```
Comment (jeffersonrabb) [inline includes/class-newspack-newsletters.php:457]:
```suggestion
				__( 'Please select a newsletter service provider.', 'newspack-newsletters' )
```

File: includes/service-providers/constant_contact/class-newspack-newsletters-constant-contact.php [hunk 2]
```diff
@@ -48,18 +69,26 @@ public function api_key() {
...
+	 * @param object $credentials API credentials.
 	 */
-	public function set_api_key( $key ) {
-		// TODO: UI for user input of Constant Contact API credentials.
-		return null;
+	public function set_api_credentials( $credentials ) {
+		if ( empty( $credentials['api_key'] ) || empty( $credentials['access_token'] ) ) {
+			return new WP_Error(
+				'newspack_newsletters_invalid_keys',
+				__( 'Please input an API key and access token.', 'newspack-newsletters' )
```
Comment (jeffersonrabb) [inline includes/service-providers/constant_contact/class-newspack-newsletters-constant-contact.php:85]:
```suggestion
				__( 'Please input Constant Contact API key and access token.', 'newspack-newsletters' )
```

File: src/components/init-modal/screens/api-keys/index.js [hunk 3]
```diff
@@ -59,87 +63,149 @@ export default ( { onSetupStatus } ) => {
...
 		<Fragment>
 			<div className={ classes }>
 				<div className="newspack-newsletters-modal__settings-wrapper">
 					{ inFlight && <Spinner /> }
-					<h4>{ __( 'Enter your Mailchimp API key', 'newspack-newsletters' ) }</h4>
-					<TextControl
-						label={ __( 'Mailchimp API key', 'newspack-newsletters' ) }
-						value={ mailchimpAPIKey }
-						onChange={ value => setKeys( { ...keys, mailchimp_api_key: value } ) }
+					<h4>{ __( 'Select your newsletter service provider', 'newspack-newsletters' ) }</h4>
```
Comment (jeffersonrabb) [inline src/components/init-modal/screens/api-keys/index.js:112]:
```suggestion
					<h4>{ __( 'Select your email service provider', 'newspack-newsletters' ) }</h4>
```

File: src/components/init-modal/screens/api-keys/index.js [hunk 4]
```diff
@@ -59,87 +63,149 @@ export default ( { onSetupStatus } ) => {
...
+								<ExternalLink href="https://mailchimp.com/help/about-api-keys/">
+									{ __( 'About Mailchimp API keys', 'newspack-newsletters' ) }
+								</ExternalLink>
+							</p>
+							<hr />
+						</Fragment>
+					) }
+					{ 'constant_contact' === serviceProvider && (
+						<Fragment>
+							<h4>{ __( 'Enter your Constant Contact API key', 'newspack-newsletters' ) }</h4>
```
Comment (jeffersonrabb) [inline src/components/init-modal/screens/api-keys/index.js:161]:
```suggestion
							<h4>{ __( 'Enter your Constant Contact API key and access token', 'newspack-newsletters' ) }</h4>
```

File: src/components/init-modal/screens/api-keys/index.js [hunk 5]
```diff
@@ -59,87 +63,149 @@ export default ( { onSetupStatus } ) => {
...
+							<TextControl
+								label={ __( 'Constant Contact API key', 'newspack-newsletters' ) }
+								value={ credentials.api_key }
+								onChange={ setCredentials( 'api_key' ) }
+								disabled={ inFlight }
+								onKeyDown={ handleKeyDown }
+								className={ errors.newspack_newsletters_invalid_keys && 'has-error' }
+							/>
+							<TextControl
+								label={ __( 'Constant Contact API Access token', 'newspack-newsletters' ) }
```
Comment (jeffersonrabb) [inline src/components/init-modal/screens/api-keys/index.js:171]:
```suggestion
								label={ __( 'Constant Contact Access token', 'newspack-newsletters' ) }
```

File: src/components/init-modal/screens/api-keys/index.js [hunk 6]
```diff
@@ -59,87 +63,149 @@ export default ( { onSetupStatus } ) => {
...
+								<p className="error">{ errors.newspack_newsletters_invalid_keys }</p>
+							) }
+
+							<p>
+								<ExternalLink href="https://constantcontact.mashery.com/">
+									{ __( 'Get Constant Contact API key', 'newspack-newsletters' ) }
+								</ExternalLink>
+								<span className="separator"> | </span>
+								<ExternalLink href="https://constantcontact.mashery.com/io-docs">
+									{ __( 'Get Constant Contact API access token', 'newspack-newsletters' ) }
```
Comment (jeffersonrabb) [inline src/components/init-modal/screens/api-keys/index.js:188]:
```suggestion
									{ __( 'Get Constant Contact access token', 'newspack-newsletters' ) }
```

File: src/components/init-modal/index.js [hunk 7]
```diff
@@ -24,12 +24,12 @@ export default ( { hasKeys, onSetupStatus } ) => {
 			shouldCloseOnClickOutside={ false }
 			shouldCloseOnEsc={ false }
 			title={
-				hasKeys
-					? __( 'Select a layout for your newsletter', 'newspack-newsletters' )
-					: __( 'Configure the newsletters', 'newspack-newsletters' )
+				shouldDisplaySettings
+					? __( 'Configure the newsletters', 'newspack-newsletters' )
```
Comment (jeffersonrabb) [inline src/components/init-modal/index.js:28]: "Configure the newsletters" has bugs me a little. What about simply "Get started"? Or "Configure plugin"?

**Comment by adekbadek** [comment]
> After completing the set up modal I'm taken to a blank editor, as opposed to the template picker as expected.

Fixed in a87df2c836b9be80c4cdba6fca0df2dc34560678 – using WP 5.5 features on day one 😎

**Pull request merged**